addons/point_of_sale/models/pos_order.py

```diff
@@ -952,6 +1005,38 @@ def _launch_stock_rule_from_pos_order_lines(self):
                 pickings_to_confirm.action_confirm()
         return True
 
+    def _is_product_storable_fifo_avco(self):
+        self.ensure_one()
+        return self.product_id.type == 'product' and self.product_id.cost_method in ['fifo', 'average']
+
+    def _compute_total_cost(self, stock_moves):
+        """
+        Compute the total cost of the order lines.
+        :param stock_moves: recordset of `stock.move`, used for fifo/avco lines
+        """
+        for line in self.filtered(lambda l: not l.is_total_cost_computed):
+            if not line.product_id:
+                line.total_cost = 0
+            else:
+                product = line.product_id
+                product_cost = line.product_id._compute_average_price(0, line.qty, stock_moves.filtered(lambda ml: ml.product_id == product)) \
```

Comment:
just to easily read it, don't use inline if.

just do:
```suggestion
                if line._is_product_storable_fifo_avco() and stock_moves: 
                    product_cost = line.product_id._compute_average_price(0, line.qty, stock_moves.filtered(lambda ml: ml.product_id == product))
                 else:
                     product_cost = line.product_id.standard_price
```

Reply:
Done, I also started to actually use (idk why i didn't before) the `product` variable actively instead of always `line.product_id`...